src/patients/ContactInfo.tsx

```diff
@@ -0,0 +1,211 @@
+      // eslint-disable-next-line react/no-array-index-key
+      <Row key={i}>
+        <Column sm={4}>
+          <SelectWithLabelFormGroup
+            name={`${name}Type${i}`}
+            value={entry.type}
+            isEditable={isEditable}
+            options={typeOptions}
+            onChange={
+              onChange
+                ? (event) => {
+                    // eslint-disable-next-line no-shadow
+                    const newData = data.map(({ value, type }) =>
+                      value === entry.value
+                        ? { value, type: event.currentTarget.value }
+                        : { value, type },
+                    )
+                    onChange(newData)
+                  }
+                : undefined
+            }
+          />
+        </Column>
+        <Column sm={8}>
+          {['tel', 'email'].indexOf(type) > -1 ? (
+            <TextInputWithLabelFormGroup
+              name={`${name}${i}`}
+              value={entry.value}
+              isEditable={isEditable}
+              onChange={
+                onChange
+                  ? (event) => {
+                      onChangeValue(event, entry.value)
+                    }
+                  : undefined
+              }
+              feedback={error}
+              isInvalid={!!error}
+              type={type}
+            />
+          ) : (
+            <TextFieldWithLabelFormGroup
+              name={`${name}${i}`}
+              value={entry.value}
+              isEditable={isEditable}
+              onChange={
+                onChange
+                  ? (event) => {
+                      onChangeValue(event, entry.value)
+                    }
+                  : undefined
+              }
+              feedback={error}
+              isInvalid={!!error}
+            />
+          )}
+        </Column>
+      </Row>
+    )
+  })
+
+  const onClickAdd = () => {
+    if (!isValid || !onChange) {
+      return
+    }
+
+    // 1. pick up only non-empty string
+    const newData = data.filter(({ value }) => value.trim() !== '')
+
+    // 2. gather errors
+    const newTempErrors = newData.map(({ value }) => (isValid(value) ? '' : errorMessageLabel))
+
+    // 3. if no error, add a new entry
+    if (newTempErrors.filter((error) => error !== '').length === 0) {
+      newData.push({ value: '' })
+    }
+
+    // 4. send updates
+    onChange(newData)
+    setTempErrors(newTempErrors)
+  }
+
+  const addButton = (
+    <div className="text-right">
+      <button
+        type="button"
+        className="btn btn-link"
+        onClick={isValid && onChange ? onClickAdd : undefined}
+      >
+        <Icon icon="add" /> {addLabel}
+      </button>
+    </div>
+  )
+
+  return isEditable && data.length === 0 ? (
+    <Spinner color="blue" loading size={20} type="SyncLoader" />
+  ) : (
+    <div>
+      {data.length > 0 ? header : null}
+      {entries}
+      {isEditable ? addButton : null}
+    </div>
+  )
+}
```

Comment:
using javascript conditionals, rather than JSX may clean this code up a bit: 


```ts
if(isEditable && data.length === 0) {
	return <Spinner color="blue" loading size={20} type="SyncLoader" />
}

return (
    <div>
      {data.length > 0 ? header : null}
      {entries}
      {isEditable ? addButton : null}
    </div>
  )
```

Reply:
✅